docs/design/generics/details.md

```diff
@@ -2505,8 +2505,24 @@ binary operator:
 -   _Same-type constraints_: `where`...`==`...
 -   _Implements constraints_: `where`...`impls`...
 
+And there are two positions that `where` can be written:
```

Comment:
I've been discussing this with @zygoloid , and we're ending up in a slightly different position. The right of an `impl as` declaration can have any `where` constraint. Rewrite constraints are the only thing that can assign values to associated constants. Other constraints are allowed, but would not normally be written. They can arise as part of using a named constraint, though, and when they do they have a meaning: if they are not satisfied by the impl declaration, a diagnostic will be issued. They don't otherwise enable anything in the impl, so it doesn't accomplish anything to write them.

Example of using a named constraint:

```
interface Container {
  let IteratorType:! Iterator;
}
constraint RandomContainer {
  extend Container where .IteratorType impls RandomAccessIterator;
}
// Checks at compile time that `MyIteratorType impls RandomAccessIterator`.
impl MyContainer as RandomContainer where .IteratorType = MyIteratorType {
  // ...
}
```

Reply:
Do you think it would be fair to describe the named constraint case as a separate 3rd position, as it has different meaning there? Or is it important to note here that you could write that type of constraint on a class impl-as as well?

Reply:
The design intent is that the language avoids distinguishing any kind of facet type from any other. This both simplifies the language and allows some evolution transitions to occur in a compatible way.